src/mongo/connectToMongoClient.ts

```diff
@@ -15,6 +15,11 @@ export async function connectToMongoClient(connectionString: string, appName: st
         useNewUrlParser: true
     };
 
+    if (!!connectionString.match(/^mongodb:\/\/(localhost|127\.0\.0\.1)/)) {
+        // Prevents self signed certificate error https://github.com/microsoft/vscode-cosmosdb/issues/1241#issuecomment-614446198
+        options.tlsAllowInvalidCertificates = true;
```

Comment:
I'm probably fine with this, but I'd like to see a more detailed explanation when you're loosening security-related settings. Basically, convince me that you're not increasing risk for our users

Reply:
Mongo recommends that valid certificates should definitely be used in production environments ([source](https://docs.mongodb.com/manual/tutorial/configure-ssl/)). But since the emulator is likely being used for development, it seems fine to allow invalid certificates here. 

Would it be better to throw up a warning before setting `tlsAllowInvalidCertificates`? Just so this isn't happening entirely under the covers

Reply:
Well why do invalid certificate errors happen? I feel like I've never run into this error and I've used a local mongo instance - can you repro?

Reply:
Yeah I was able to repro along with Nathan & CTI. It looks like certificate behavior changed in mongo v4.0: https://docs.mongodb.com/manual/release-notes/4.0/#x-509-authentication-certificate-restrictions. So it makes sense that we're just now seeing this type of error

Reply:
Are you reproducing with the Cosmos DB emulator for mongo or just plain old mongodb? If it only repros in the Cosmos DB case, we might not want to do this for everyone

Reply:
I've only ever seen the error when using the emulator. Are you suggesting we only set this flag if the connection string is for the Cosmos DB emulator?

Reply:
Yeah